src/persist-client/src/impl/state.rs

```diff
@@ -229,6 +249,16 @@ where
             .expect("TODO: Implement automatic lease renewals")
     }
 
+    fn writer(&mut self, id: &WriterId) -> &mut WriterState {
+        self.writers
+            .get_mut(id)
+            // The only (tm) ways to hit this are (1) inventing a WriterId
+            // instead of getting it from Register or (2) if a lease expired.
+            // (1) is a gross mis-use and (2) isn't implemented yet, so it feels
```

Comment:
> (2) isn't implemented yet

lol

Reply:
heh, I read this as still being partially true. we renew leases while one is active, but we still don't have a contingency for renewing a WriteHandle (maybe by re-registering with a new write ID) after it's expired. I'll update this message + error though since it's confusing